client/lecturer/app/engagement.directive.js

```diff
@@ -19,9 +19,32 @@
         function link(scope, element, attrs, engagementCanvas) {
             console.log('Ready (Engagement Canvas link)');
 
-            engagementCanvas.canvas = element[0];
+            engagementCanvas.width = element.width();
+            engagementCanvas.height = element.height();
+
+            engagementCanvas.canvas = document.createElement('canvas');
+            element.append(engagementCanvas.canvas);
             engagementCanvas.ctx = engagementCanvas.canvas.getContext('2d');
 
+            engagementCanvas.canvas.width = engagementCanvas.width;
+            engagementCanvas.canvas.height = engagementCanvas.height;
```

Comment:
This was a bit confusing to me. Is it necessary having both  `engagementCanvas.width` and engagementCanvas.canvas.width`?
In other words change it to `engagementCanvas.canvas.width = element.width(); ` maybe?